packages/flame/lib/src/sprite_animation.dart

```diff
@@ -300,6 +306,9 @@ class SpriteAnimation {
       return;
     }
     while (clock >= currentFrame.stepTime) {
+      if (isFirstFrame){
+        onStart?.call();
```

Comment:
This will be called several times if the animation is looping, is that the intention?

Reply:
What about if we had a `onFrame(frameIndex)` instead, so we can have the flexibility for having a callback on whatever frames they want.

Reply:
Yes.  I figured if say you had like a clock animation or such and you wanted to trigger something at the start of the animation instead of at the end, this would accomplish that.

Reply:
> What about if we had a `onFrame(frameIndex)` instead, so we can have the flexibility for having a callback on whatever frames they want.

So would you deprecate onComplete as well? I do like the onFrame concept.

Reply:
`onComplete` is not being called every loop, it is only running once the animation is "done", so it should not be deprecated.

Reply:
Wait a second... Is the intention here that `onStart` should be called after the first frame has finished, because that is what it seems to do now? Shouldn't it be called as soon as it starts the first frame?

Reply:
Yes, so seeking clarification, if `isFirstFrame` is 0, is that after the first frame has rendered or before it has rendered?